secio/src/peer_id.rs

```diff
@@ -57,8 +57,8 @@ impl PeerId {
 
     /// Return `PeerId` which used hashed seed as inner.
     fn from_seed(seed: &[u8]) -> Self {
-        let mut buf = encode::u16_buffer();
-        let code = encode::u16(SHA256_CODE, &mut buf);
+        let mut buf = encode::u64_buffer();
+        let code = encode::u64(SHA256_CODE, &mut buf);
```

Comment:
`encode::u16_buffer()` will got a `[0; 3]` buf,
`encode::u64_buffer()` will got a `[0; 10]` buf.

So `code` will be different length, `inner` will be different too, finaly `PeerId { inner }` will be different.

Reply:
```rust
    /// Generate Peer id
    pub fn peer_id(&self) -> PeerId {
        self.public_key().peer_id()
    }
```
Will this change cause same `public_key`'s `self.public_key().peer_id()`got different peer_id?

Reply:
https://github.com/paritytech/unsigned-varint/blob/master/src/decode.rs#L66-L86

Although the bytes data has changed, its decode is consistent

Reply:
OK, confirmed.

Reply:
> ```rust
>     /// Generate Peer id
>     pub fn peer_id(&self) -> PeerId {
>         self.public_key().peer_id()
>     }
> ```
> 
> Will this change cause same `public_key`'s `self.public_key().peer_id()`got different peer_id?

https://github.com/paritytech/unsigned-varint/blob/master/src/encode.rs#L22-L38
 After encoding, no matter the context is [0;10] or [0;13], the resulting code is [18]